src/test/java/org/terasology/module/behaviors/MovementTests.java

```diff
@@ -325,30 +325,38 @@ void cleanUp() {
     }
 
     private EntityRef createMovingCharacter(float height, float radius, Vector3i start, Vector3i stop, String... movementTypes) {
-        EntityRef entity = entityManager.create("Behaviors:testcharacter");
-        entity.send(new CharacterTeleportEvent(new Vector3f(start)));
-
-        MinionMoveComponent minionMoveComponent = new MinionMoveComponent();
-        minionMoveComponent.setPathGoal(stop);
-        minionMoveComponent.movementTypes.addAll(Sets.newHashSet(movementTypes));
-        entity.addOrSaveComponent(minionMoveComponent);
+        EntityBuilder builder = entityManager.newBuilder("Behaviors:testCharacter");
+        builder.setSendLifecycleEvents(true);
+        builder.upsertComponent(MinionMoveComponent.class, maybeComponent -> {
+            MinionMoveComponent moveComponent = maybeComponent.orElse(new MinionMoveComponent());
+            moveComponent.setPathGoal(stop);
+            moveComponent.movementTypes.clear();
+            moveComponent.movementTypes.addAll(Sets.newHashSet(movementTypes));
+            return moveComponent;
+        });
+        builder.updateComponent(CharacterMovementComponent.class, characterMovement -> {
```

Comment:
I wonder whether we should use `upsertComponent` all the time, or at least log something if the component is not present? 🤔 Should `updateComponent` itself log something (DEBUG level?) if the component is not present? 

The main drawback of using `updateComponent` is that it will silently do nothing if the component is not present. So, `upsertComponent` is the safe bet to ensure the right thing will happen, while `updateComponent` is a bit nicer to read if you're pretty certain that the component will be present...

Reply:
Not sure whether I would encourage using `updateComponent` only because it's more readable but in some cases doesn't do what I want it to do based on an assumption I had that might have changed by now...
I guess `upsertComponent` might indeed be better here.